gramps/gen/filters/rules/person/_hascommonancestorwith.py

```diff
@@ -96,17 +106,22 @@ def add_ancs(self, db, person):
     def reset(self):
         self.ancestor_cache = {}
 
-    def has_common_ancestor(self, other):
+    def has_common_ancestor(self, other: Person):
         for handle in self.with_people:
-            if (handle in self.ancestor_cache and self.ancestor_cache[handle]) & (
+            # Requires bitwise and for some reason:
```

Comment:
Did you investigate why?

Reply:
No, I only verified that it needed to be that way.

Reply:
```suggestion
```
this is actually doing intersection between two dicts